Change console log line number color

tlmii opened the pull request:
Resolves #2842 

After chatting with @leslierichardson95, here's where I landed:

Before:
![image](https://github.com/dotnet/aspire/assets/9613109/2813a186-70e8-4aaf-92d3-6ac72639d501)
After:
![image](https://github.com/dotnet/aspire/assets/9613109/bef15024-5094-4f73-827f-32887cb0c815)

That's about as light as we can go for the line numbers while maintaining contrast. I also changed the timestamp to be brighter, like in Docker Desktop (though not as bright), rather than dimmer. 

I _think_ I preferred the timestamp being dimmed to keep it separate from the main content... but I do like having the line numbers dimmed.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/3035)

tlmii committed:
Change console log line number color

JamesNK commented:
> I think I preferred the timestamp being dimmed to keep it separate from the main content... but I do like having the line numbers dimmed.

Timestamp should be dimmed. It's not as important as other content and darker color helps focus on other content.

IMO timestamp and line number should both be dimmed but different amounts.

tlmii commented:
> IMO timestamp and line number should both be dimmed but different amounts.

There's not a lot of space for two noticeably different dimmed amounts before we fail contrast checks. But I'll give it a shot. Looking at the before/after more I really do think the dimmed timestamp focuses the important content a lot better.

JamesNK commented:
Could also consider increasing the spacing between line numbers and content. VS code for example:

![image](https://github.com/dotnet/aspire/assets/303201/692cef72-7e14-4cf8-b7e7-2b7e88287627)

They don't need to be so close together.

drewnoakes commented:
I like the line after screenshot a lot.

While we're at it, why not trim the date off the front of the value if it's today's date (in local time)? We do this in the resource grid, and it makes the UI a bit cleaner.

drewnoakes commented:
> Could also consider increasing the spacing between line numbers and content.

The font size could be a little larger too, IMO.

ilmax commented:
> I like the line after screenshot a lot.
> 
> While we're at it, why not trim the date off the front of the value if it's today's date (in local time)? We do this in the resource grid, and it makes the UI a bit cleaner.

I'm on the other side of the spectrum and this seems like a regression to me, 99% of the time I don't care about time until I find what I'm looking for in the log messages

tlmii committed:
PR feedback, organization, hover, tweaks

tlmii commented:
OK, here's another stab at it.

- Increased the spacing between the line number of the timestamp/content to match AzDO
- Added hover color to line background similar to AzDO/GH logs and tweaked the line number color slightly for valid contrast
- Changed the timestamp back to dimmed, same color as line number
- Reorganized all the "console theme" colors to into variables to make it easier to modify them (and, theoretically, to make it easier to have actual console themes in the future)

Original before:
![image](https://github.com/dotnet/aspire/assets/9613109/a3866c3c-4293-4719-b148-2b968a78f225)
First attempt above:
![image](https://github.com/dotnet/aspire/assets/9613109/21f6d02c-f592-4582-b185-baa9193c39eb)
After:
![image](https://github.com/dotnet/aspire/assets/9613109/5df73305-2f74-4d52-b0bc-8da93adf62ab)

I _think_ this works for me. Separation between line number and text helps to differentiate the actual log entry from the line number. Dimmed timestamp keeps the focus on the content. Background highlight on hover helps with the recognition of individual log entries.

davidfowl commented:
/azp run
